### ISSUE TITLE
Should set() and delete() promises have return values?

### ISSUE THREAD

**Issue by domenic** [domenic/async-local-storage#7]
Right now set() and delete() return promises for undefined. This is the most simple thing that falls out of IndexedDB semantics.

We could try to adhere closer to JS Map semantics, and have set() return a promise for the value that was set (which in our case is actually a _clone_ of the value that was passed), and have delete() return a promise for a boolean indicating whether or not anything was deleted.

Doing so would require extra operations on the backing database, and doesn't seem that useful. Also, this should be a backward-compatible change to add if someone has a very compelling use case. So, I'm tentatively resolved to leave these promises as undefined. But, I want to leave this issue open to indicate I'm open to changing our position.

**Comment by domenic** [comment]
And if it did, what would be "the set value"? A clone of the set value (i.e. what was actually stored), or just the value passed in?

Relatedly, should delete() return a promise for a boolean indicating whether or not a deletion happened? (Like JS Maps do.)

Initially I said yes to this latter question, but when I went to spec it in https://github.com/domenic/async-local-storage/commit/19bc9b4f381b2314875d94f45063e0a9ce9cc86b I removed this since it did not fall out naturally.

I'm currently thinking we should just do what's natural for async local storage/IDB, and not ape these exact details of JS's Map class when it would require extra work (i.e. extra roundtrips to the database). Thoughts welcome though.
